### DISCUSSION TITLE
Inconsistent Split Navigation Behavior

### DISCUSSION THREAD

**Discussion by 766F6964** [ghostty-org/ghostty#3604] [Q&A] [answered]
# Current Behaviour

When navigating between splits using directional commands (e.g., goto_split:left), the navigation doesn't consistently select the visually adjacent split in the specified direction. 
Instead, it appears to select splits based on creation order rather than spatial position.

## Steps to Reproduce

1. Launch ghostty with the following configuration:

```
# Creating splits
keybind = ctrl+shift+h=new_split:left
keybind = ctrl+shift+j=new_split:down
keybind = ctrl+shift+k=new_split:up
keybind = ctrl+shift+l=new_split:right

# Navigating splits
keybind = ctrl+h=goto_split:left
keybind = ctrl+j=goto_split:bottom
keybind = ctrl+k=goto_split:top
keybind = ctrl+l=goto_split:right
```
2. Create the following layout:
  -  Press `ctrl+shift+l` to create a split on the right
  - Press `ctrl+shift+j` to create a split below it
  - **Result:** Three splits arranged in an "L" shape (see screenshot)

![Screenshot From 2024-12-27 23-29-39](https://github.com/user-attachments/assets/988391cb-b5e3-4281-8004-eb6c6ebfbbc5)

3. With focus in the bottom-right split, `press ctrl+h` to attempt to navigate left.

## Expected vs Actual Behavior

- **Expected:** Focus moves to the left-most split (the visually adjacent split)
- **Actual:** Focus moves to the top-right split (the previously created split)

| Expected Behaviour | Actual Behaviour |
|-------------------|-----------------|
| ![expected](https://github.com/user-attachments/assets/70feeb9f-94ee-4e38-97b7-6e91b9827d4e) | ![actual](https://github.com/user-attachments/assets/5db070a4-0da9-4a06-ac18-5dc7c549a80f) |

**Comment by rhodes-b** [comment]
> it appears to select splits based on creation order rather than spatial position

I believe this is the expect behavior

**Reply by rhodes-b** [reply, marked as answer]
https://github.com/ghostty-org/ghostty/issues/2866
this seems important

**Reply by 766F6964** [reply]
> #2866 this seems important

Thanks for linking this issue.
And I agree with the other commenters in that issue, - this does not make sense for expected behavior.
If I have a keybind set up to navigate left, I expect it to navigate left and not up, etc.

This should be fixable if every split has some sort of pointer/reference to its immediate neighbors (left, right, up, down).
These would have to be updated with every new split, or on resizing tho.
